### PR TITLE
Replace sh scripts of libs/unit with Haxe scripts

### PR DESCRIPTION
**Pull request by PXshadow**

### CODE REVIEW

File: src/Main.hx [hunk 1]
```diff
@@ -227,10 +227,8 @@ class Main {
 				if (arg.charAt(0) == '-') {
 					throw 'Unknown argument "$arg"';
 				}
-				// remove ".d.ts" extension as this is a common point of confusion for users
-				var moduleName = if (arg.substr(arg.length - 5).toLowerCase() == '.d.ts') {
-					arg.substr(0, arg.length - 5);
-				} else arg;
+				// remove ".d.ts" extension as this is a common point of confusion for users, (remove extension for moduleName to check against module path lookup)
+				final moduleName = Path.withoutExtension(arg);
```
Comment (haxiomic) [inline src/Main.hx:231]: Unfortunately this doesn’t quiet work, extensions are defined as characters after the last . whereas here we want to include the second to last .

File: test/RunLibs.hx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+function main() {
+    Test.run('_generated-libs',"",[
+        'node',
+        'three',
```
Comment (haxiomic) [inline test/RunLibs.hx:4]: Would be useful to also do three orbit controls in the same dts2hx execution (see run lib sh)

File: src/Main.hx [hunk 3]
```diff
@@ -322,10 +320,10 @@ class Main {
 				var packageObj = haxe.Json.parse(packageJson);
 				var dependencies: DynamicAccess<String> = packageObj.dependencies != null ? packageObj.dependencies : {};
 				var devDependencies: DynamicAccess<String> = packageObj.devDependencies != null ? packageObj.devDependencies : {};
-				var allDependencies = dependencies.keys().concat(devDependencies.keys());
```
Comment (haxiomic) [inline src/Main.hx:325]: allDependencies might be better here to remind me what this was about (moduleNames gives less info)

**Review by haxiomic — APPROVED** [review]
Hey thanks for this, looks good, much better as a haxe script but there are few little tweaks I’d suggest

**Comment by haxiomic** [comment]
Happy for me to merge ?

**Comment by PXshadow** [comment]
Yes indeed :)

**Pull request merged**